event-store/src/adapters/amqp/emitter.rs

```diff
@@ -0,0 +1,182 @@
+                    error!("Error connecting to AMQP: {}", e);
+                }),
+        );
+        let channel = rx
+            .recv_timeout(Duration::from_secs(5))
+            .map_err(|e| {
+                error!("AMQP channel was not made available: {}", e);
+                panic!(e);
+            }).expect("AMQP channel was not made available");
+        Self {
+            channel,
+            namespace,
+            exchange: _exchange,
+        }
+    }
+}
+
+fn prepare_subscription<E, H>(
+    queue_name: String,
+    event_name: String,
+    exchange: String,
+    handler: H,
+    channel: Channel<TcpStream>,
+) -> impl Future<Item = (), Error = io::Error>
+where
+    E: EventData,
+    H: Fn(&Event<E>) -> (),
+{
+    let c_channel = channel.clone();
+    let queue_name1 = queue_name.clone();
+    info!("Creating queue {}", queue_name);
+    channel
+        .queue_declare(
+            &queue_name,
+            QueueDeclareOptions {
+                durable: true,
+                exclusive: false,
+                auto_delete: false,
+                ..QueueDeclareOptions::default()
+            },
+            FieldTable::new(),
+        ).and_then(move |queue| {
+            info!("Binding queue {} to exchange {}", queue_name, exchange);
+            channel
+                .queue_bind(
+                    &queue_name,
+                    &exchange,
+                    &event_name,
+                    QueueBindOptions::default(),
+                    FieldTable::new(),
+                ).and_then(move |_| {
+                    channel.basic_consume(
+                        &queue,
+                        &queue_name,
+                        BasicConsumeOptions::default(),
+                        FieldTable::new(),
+                    )
+                }).and_then(move |stream| {
+                    info!("Starting to consume from queue {}", queue_name1);
+                    stream.for_each(move |message| {
+                        let data: Event<E> =
+                            serde_json::from_str(str::from_utf8(&message.data).unwrap()).unwrap();
+                        info!("Receiving message with id {}", data.id);
+                        handler(&data);
+                        c_channel.basic_ack(message.delivery_tag, false)
+                    })
+                })
+        }).and_then(|_| ok(()))
+        .map_err(|e| e.into())
+}
+
+impl EmitterAdapter for AMQPEmitterAdapter {
+    fn get_subscriptions(&self) -> Vec<String> {
+        vec![]
+    }
+
+    fn emit<'a, E: Events + Sync>(
+        &self,
+        event: &Event<E>,
+    ) -> Box<Future<Item = (), Error = io::Error> + Send + Sync> {
+        let payload: Vec<u8> = serde_json::to_string(event)
+            .expect("Cant serialise event")
+            .into();
+        let event_type = event.data().event_type();
+        let id = event.id;
+        info!("Emiting event {} with id {}", event_type, id);
```

Comment:
Nit: two Ts

Reply:
`panic!("Jargon not found")`

I'll need a bit more of context to resolve this. I do not understand what you mean with `Nit` and `Ts`

Reply:
`Emiting` should be `Emitting` with two Ts. `Nit` is short for `nitpick` :)